File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        List<Integer> res = new ArrayList<>();
        Arrays.sort(nums1);
        Arrays.sort(nums2);

        for(int i : nums1){
            int l = 0;
            int r = nums2.length - 1;
            while(l <= r){
                int mid = (l + r) / 2;
                if(nums2[mid] == i && !res.contains(i)){
                    res.add(i);
                }else if(nums2[mid] > i){
                    r = mid - 1;
                }else{
                    l = mid + 1;
                }
            }
        }
        int[] result = new int[res.size()];
        for(int i = 0 ; i < res.size() ; i++){
            result[i] = res.get(i);
        }
        return result;
    }
}

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        res = []
        nums1.sort()
        nums2.sort()

        for i in nums1:
            l , r = 0 , len(nums2) - 1
            while l <= r:
                mid = (r - l) // 2 + l
                if nums2[mid] == i and i not in res:
                    res.append(i)
                    
                elif nums2[mid] > i:
                    r = mid - 1
                else:
                    l = mid + 1
        return res
